pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1242,43 +1233,102 @@ func (vs *VSphere) CreateVolume(volumeOptions *VolumeOptions) (volumePath string
 		return "", err
 	}
 
-	// vmdks will be created inside kubevols directory
-	kubeVolsPath := filepath.Clean(ds.Path(VolDir)) + "/"
-	err = makeDirectoryInDatastore(vs.client, dc, kubeVolsPath, false)
-	if err != nil && err != ErrFileAlreadyExist {
-		glog.Errorf("Cannot create dir %#v. err %s", kubeVolsPath, err)
-		return "", err
-	}
-	glog.V(4).Infof("Created dir with path as %+q", kubeVolsPath)
+	// Create a disk with the VSAN storage capabilities specified in the volumeOptions.StorageProfileData.
+	// This is achieved by following steps:
+	// 1. Create dummy VM with the disk configured with VSAN policy attached.
+	// 2. Detach the disk from the dummy VM.
+	// 3. Delete the dummy VM. This removes all the vmx files.
+	// 4. Move the Virtual Disk to Kubevols directory.
+	// 5. Delete the dummy VM folder as now its contents are empty.
```

Comment:
There is no need to detach disk from dummy VM. You can attach the same disk to kubernetes node. Note that the dummy VM will never be powered on, so there is no problem.